packages/web3-validator/src/utils.ts

```diff
@@ -449,20 +449,49 @@ export function uint8ArrayToHexString(uint8Array: Uint8Array): string {
 	return hexString;
 }
 
+// for optimized technique for hex to bytes conversion
+const charCodeMap = {
+	zero: 48,
+	nine: 57,
+	A: 65,
+	F: 70,
+	a: 97,
+	f: 102,
+  } as const
+
+  function charCodeToBase16(char: number) {
+	if (char >= charCodeMap.zero && char <= charCodeMap.nine)
+	  return char - charCodeMap.zero
+	if (char >= charCodeMap.A && char <= charCodeMap.F)
+	  return char - (charCodeMap.A - 10)
+	if (char >= charCodeMap.a && char <= charCodeMap.f)
+	  return char - (charCodeMap.a - 10)
+	return undefined
+  }
```

Comment:
@mpetrunic Is this function not equivalent to the native `parseInt` with difference take it take integer parameter while `parseInt` take string parameter as `charCodeToBase16(97)` is same as `parseInt('a', 16)`.

The native `parseInt` in that use case is almost 5 times more faster. 

https://runkit.com/nazarhussain/6571df9b17589500083041e5